modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -274,6 +275,256 @@ private void removeLocalSubscriptionAndNotify(AndesSubscription subscription) th
+     *
+     * @param isDurable true if searching for durable subscriptions
+     * @param protocolType protocol of the subscription
+     * @param destinationType type of subscription (QUEUE/TOPIC/DURABLE_TOPIC)
+     * @param bindingKeyPattern regex to match with binding key of subscriber
+     * @param subscriptionIdPattern regex to match with ID of the subscriber
+     * @param isExactMatchSubscriptionId exact match of subscription id or not
+     * @return Set of subscriptions filtered according to search criteria
+     * @throws AndesException
+     */
+    private Set<AndesSubscription> getInactiveSubscriptionsByTokenizedBindingKeyMatch(boolean isDurable,
+                                                                                     ProtocolType protocolType,
+                                                                                     DestinationType destinationType,
+                                                                                     String bindingKeyPattern,
+                                                                                     String subscriptionIdPattern,
+                                                                                     boolean isExactMatchSubscriptionId)
+            throws AndesException {
+
+        Set<AndesSubscription> filteredSubscriptions = new HashSet<>();
+        String messageRouter = destinationType.getAndesMessageRouter();
+
+        List<AndesSubscription> allInactiveSubscriptions = getInactiveSubscriberRepresentations();
+
+        for (AndesSubscription inactiveSubscription : allInactiveSubscriptions) {
+            if(isExactMatchSubscriptionId && inactiveSubscription.isDurable() == isDurable
+                    && inactiveSubscription.getStorageQueue().getMessageRouter().getName().equals(messageRouter)
+                    && inactiveSubscription.getStorageQueue().getMessageRouterBindingKey().toLowerCase().
+                    contains(bindingKeyPattern.toLowerCase())
+                    && inactiveSubscription.getProtocolType().equals(protocolType)
+                    && inactiveSubscription.getSubscriptionId().equalsIgnoreCase(subscriptionIdPattern)) {
+                filteredSubscriptions.add(inactiveSubscription);
+            }else if(inactiveSubscription.isDurable() == isDurable
+                    && inactiveSubscription.getStorageQueue().getMessageRouter().getName().equals(messageRouter)
+                    && inactiveSubscription.getStorageQueue().getMessageRouterBindingKey().toLowerCase().
+                    contains(bindingKeyPattern.toLowerCase())
+                    && inactiveSubscription.getProtocolType().equals(protocolType)
+                    && inactiveSubscription.getSubscriptionId().toLowerCase().contains(subscriptionIdPattern
+                    .toLowerCase())){
+                filteredSubscriptions.add(inactiveSubscription);
+            }
+        }
+        return filteredSubscriptions;
+    }
+
+
+
+
```

Comment:
After the introduction of these new methods the number of lines in this class has reached close to 1000. Consider refactoring.